### ISSUE TITLE
JSpecify: false positive on array of nullable elements of type with generic

### ISSUE THREAD

**Issue by unv-unv** [uber/NullAway#1800]
```java
@NullMarked
public static class Test<E> {
    public void foo(@Nullable Test[] array) {
        // BUG: 
        array[0] = null;
    }
}
```

Shows compilation error:
```
[ERROR] Test.java:[5,21] [NullAway] Writing @Nullable expression into array with @NonNull contents.
[ERROR]     (see http://t.uber.com/nullaway )
```

JSpecify mode is on.

NullAway 0.14.1, Error Prone 2.50.0.

Note that the same example works if I remove <E>:

```java
public static class Test {
    public void foo(@Nullable Test[] array) {
        array[0] = null;
    }
}
```
